### ISSUE TITLE
useRouteQuery: Can't update multiple route query in the same tick

### ISSUE THREAD

**Issue by peter50216** [vueuse/vueuse#901]
CodeSandbox example: https://codesandbox.io/s/vueuse-useroutequery-same-tick-i739c

Currently the route query update in useRouteQuery is updating with [the old `route.query`](https://github.com/vueuse/vueuse/blob/main/packages/router/useRouteQuery/index.ts#L27), which means that if in the same tick we try to set value of two route query (for example, swapping two values), only one of them would work.

**Comment by Shinigami92** [comment]
I can reproduce this

**Comment by pipinet** [comment]
I can reproduce this too

**Comment by MaicoTimmerman** [comment]
I'm running into the same issue, any suggestions for a workaround?

**Comment by Shinigami92** [comment]
@MaicoTimmerman 

I currently have this in my project, but it's not battle-tested!

```ts
import type { MaybeRef } from '@vueuse/core';
import type { UnwrapNestedRefs } from 'vue';
import { reactive, unref, watch } from 'vue';
import type { LocationQueryValue } from 'vue-router';
import { useRoute, useRouter } from 'vue-router';

export interface RouteQueriesOptions {
  /**
   * Mode to update the router query, ref is also acceptable
   *
   * @default 'replace'
   */
  mode?: MaybeRef<'replace' | 'push'>;

  /**
   * Route instance, use `useRoute()` if not given
   */
  route?: ReturnType<typeof useRoute>;

  /**
   * Router instance, use `useRouter()` if not given
   */
  router?: ReturnType<typeof useRouter>;
}

export function useRouteQuery<
  T extends Record<string, LocationQueryValue | LocationQueryValue[]>,
>(
  defaults: T,
  {
    mode = 'replace',
    route = useRoute(),
    router = useRouter(),
  }: RouteQueriesOptions = {},
): UnwrapNestedRefs<T> {
  const query = reactive<T>({
    ...Object.fromEntries(
      Object.entries(defaults).map(([key, value]) => [
        key,
        route.query[key] || value,
      ]),
    ),
  } as T);

  watch(
    query,
    async (value) => {
      await router[unref(mode)]({
        query: {
          // ...$route.query,
          ...value,
        },
      });
    },
    { immediate: true },
  );

  return query;
}
```

**Comment by renatodeleao** [comment]
Can also reproduce, trying to set `daterange` query params (`starts_at`, `ends_at`) => [demo](https://codesandbox.io/s/useroutequery-not-updating-multiple-params-8yr6fi). 
Currently hacking by moving one of the setters to the `nextTick` (wrapping within setTimeout, but not an actual fix).

**Comment by denishacquin** [comment]
Same issue here, would love a fix

**Comment by WilcoKruijer** [comment]
Not stale, please reopen. (I saw this 16 hours too late, sadly)

**Comment by phillnom** [comment]
Still an issue for me.

**Comment by sean-mllrdev** [comment]
Still an issue for me

**Comment by andrey-hohlov** [comment]
Don't want to drop this composable. Hopefully PR with fix will be merged soon.

**Comment by wheatjs** [comment]
This is fixed in the latest version [v9.11.0](https://github.com/vueuse/vueuse/releases/tag/v9.11.0)
